Add "docker.io/" registry prefix to docker compose image

Valloric opened the pull request:
Required by Podman Compose. Docker Compose should not care either way. The registry prefix is explicitly supported in the [compose file spec](https://github.com/compose-spec/compose-spec/blob/master/spec.md#image)

<!-- Thank you for submitting a pull request! Make sure to follow the instructions to write release notes for your PR — it should only take a minute or two: https://github.com/actualbudget/docs#writing-good-release-notes -->

joel-jeremy commented:
@MatissJanis do you think we should point the docker compose file to the GitHub container registry or should we use docker hub instead?

joel-jeremy commented:
@Valloric actual server images are also hosted on docker hub. Can we point the compose file there instead?

Valloric commented:
I changed the prefix to point to docker hub and added the release notes. Should be good to go now.

Some friendly advice from someone who spent 15+ years maintaining various OSS projects: the [release notes process](https://actualbudget.org/docs/contributing/#writing-good-release-notes) adds needless overhead and will cut the rate of PRs you get by a factor of 10. This process is largely automatable at the time of cutting a release. The non-automatable parts (categorization of bugfix/feature/whatever) is not worth the overhead.

Do with that advice as you wish. :)

joel-jeremy commented:
Thank you! We'll keep that advice in mind. :)
